flaml/integrations/oai/completion.py

```diff
@@ -0,0 +1,441 @@
+                    x = data[i]
+                    responses = responses_list[i]
+                    metrics = cls._eval_func(responses, **x)
+                    if result:
+                        for key, value in metrics.items():
+                            result[key] += value
+                    else:
+                        result = metrics
+                for key in result.keys():
+                    result[key] /= data_limit
+                result["total_cost"] = cls._total_cost
+                result["cost"] = cost
+                result["inference_cost"] = (
+                    n_tokens * cls.price1K[config["model"]] / 1000
+                )
+                break
+            else:
+                if data_early_stop:
+                    previous_n = 0
+                    n_tokens_list.clear()
+                    responses_list.clear()
+                else:
+                    previous_n = n
+                n = min(n << 1, config_n)
+        return result
+
+    @classmethod
+    def tune(
+        cls,
+        data,
+        metric,
+        mode,
+        eval_func,
+        log_file_name=None,
+        inference_budget=None,
+        optimization_budget=None,
+        num_samples=1,
+        **config,
+    ):
+        """Tune the parameters for the OpenAI API call.
+
+        Args:
+            data (list): The list of data points.
+            metric (str): The metric to optimize.
+            mode (str): The optimization mode, "min" or "max.
+            eval_func (Callable): The evaluation function for responses.
+            log_file_name (str, optional): The log file.
+            inference_budget (float, optional): The inference budget.
+            optimization_budget (float, optional): The optimization budget.
+            num_samples (int, optional): The number of samples to evaluate.
+            **config (dict): The search space to update over the default search.
+                For prompt, please provide a string or a list of strings.
+                For stop, please provide a string, a list of strings, or a list of lists of strings.
+
+        Returns:
+            dict: The optimized hyperparameter setting.
+            tune.ExperimentAnalysis: The tuning results.
+        """
+        space = Completion.default_search_space.copy()
+        if config is not None:
+            space.update(config)
+            temperature = space.pop("temperature", None)
+            top_p = space.pop("top_p", None)
+            if temperature is not None and top_p is None:
+                space["temperature_or_top_p"] = {"temperature": temperature}
+            elif temperature is None and top_p is not None:
+                space["temperature_or_top_p"] = {"top_p": top_p}
+            elif temperature is not None and top_p is not None:
+                space.pop("temperature_or_top_p")
+                space["temperature"] = temperature
+                space["top_p"] = top_p
```

Comment:
not recommended by openai

Reply:
Yes. This is when users specify temperature and top_p together. What's your suggestion?

Reply:
print a warning?